Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationJsonWriter.cs

```diff
@@ -31,22 +31,30 @@ public void ApplyReport(AnnotationReport report)
             
             foreach (var lambdaFunction in report.LambdaFunctions)
             {
-                ProcessLambdaFunction(lambdaFunction);
-                processedLambdaFunctions.Add(lambdaFunction.Name);
+                if (ProcessLambdaFunction(lambdaFunction))
+                    processedLambdaFunctions.Add(lambdaFunction.Name);
             }
 
             RemoveOrphanedLambdaFunctions(processedLambdaFunctions);
             _fileManager.WriteAllText(report.CloudFormationTemplatePath, _jsonWriter.GetPrettyJson());
         }
         
-        private void ProcessLambdaFunction(ILambdaFunctionSerializable lambdaFunction)
+        private bool ProcessLambdaFunction(ILambdaFunctionSerializable lambdaFunction)
         {
             var lambdaFunctionPath = $"Resources.{lambdaFunction.Name}";
             var propertiesPath = $"{lambdaFunctionPath}.Properties";
-            
+
             if (!_jsonWriter.Exists(lambdaFunctionPath))
+            {
                 ApplyLambdaFunctionDefaults(lambdaFunctionPath, propertiesPath);
-
+            }
+            else
+            {
+                var creationTool = _jsonWriter.GetToken($"{lambdaFunctionPath}.Metadata.Tool", string.Empty);
+                if (!string.Equals(creationTool.ToObject<string>(), "Amazon.Lambda.Annotations", StringComparison.Ordinal))
+                    return false;
+            }
+            
```

Comment:
Extract the logic to check whether a lambda function can be mutated to separate method. `ProcessLambdaFunction` must not be called if metadata is not expected.

something like in L34.
```csharp
if (GetToolMetadata(arg1, ag2, ...) != "Amazon.Lambda.Annotations") 
{
     continue;
}

ProcessLambdaFunction(arg1)
```

Reply:
Fixed